docs/userguides/compile.md

```diff
@@ -81,3 +81,25 @@ Alternatively, configure it to always happen:
 compile:
   use_dependencies: true
 ```
+
+## Settings
+
+Generally, configure compiler plugins using your `ape-config.yaml` file.
+For example, when using the `vyper` plugin, you can configure settings under the `vyper` key:
+
+```yaml
+vyper:
+  version: 0.3.10
+```
+
+You can also configure adhoc settings in Python code:
+
+```python
+from ape import compilers
+
+compilers.compile("path/to/contract.vy", settings={"version": "0.3.7"})
```

Comment:
**issue**: if using top level manager, the compiler name key should also be in the dict so you can set settings for multiple compilers if needbe


- [ ] Change everywhere